Sources/TransactionalProperty.swift

```diff
@@ -0,0 +1,690 @@
+import Result
+
+/// A mutable, observable property that has an optionally failable action
+/// associated with the setter.
+///
+/// ## Consistency when nested
+///
+/// `TransactionalProperty` would back-propagate the validation failure if a
+/// proposed value from an outer property fails with its inner property.
+///
+/// It would also evaluate values originated from the inner property. In other
+/// words, it is possible for `value` to be invalid, causing `validations`
+/// to be a `failure`. Rely on `validations` for asserting a pass in validation.
+public final class TransactionalProperty<Value, TransactionError: Error>: ComposableMutablePropertyProtocol {
```

Comment:
Love the functionality, but not the name.

I don't sense anything that feels like a "transaction" when reading examples of using the class. Instead, I think that something like `PropertyEditor`, or the previously used `EditableProperty`, might better explain what the class does. Also, I find that the mixing of a "forward `map`" and `transform` (i.e. the `init` parameters vs the `map(forward:backward:)`) is a little confusing.

The thing that feels "weird" about `PropertyEditor`, though is that it behaves just like a property. So if I have a variable like `emailEditor` then it'd be weird to ask for `emailEditor.value` or `emailEditor.producer`.

Another idea that sprung up just now is a `{Mutable,}PropertyTransformer`, or maybe even `MutablePropertyAdapter` (with the former being my preferred of the two.) Those would tie in nicely with the transformation nomenclature that's used in places.